packages/devtools/lib/src/charts/flame_chart_canvas.dart

```diff
@@ -33,44 +32,39 @@ const _shadedBackgroundColor =
 
 const _fontSize = 14.0;
 const _textOffsetY = 18.0;
-const _flameChartTop = rowHeightWithPadding;
-const _rowHeight = 25.0;
-const _rowPadding = 2.0;
-const rowHeightWithPadding = _rowHeight + _rowPadding;
-
-const _flameChartInset = 70;
+const rowPadding = 2.0;
+const rowHeightWithPadding = flameChartRowHeight + rowPadding;
+const flameChartTop = rowHeightWithPadding;
+const flameChartRowHeight = 25.0;
+const flameChartInset = 70;
 
 List<num> _asciiMeasurements;
 
-// TODO(kenzie): move this class to flame_chart.dart once the frame flame chart
-// is ported to canvas and the current implementation in flame_chart.dart is
-// deleted.
-abstract class FlameChart {
+abstract class FlameChart<T> {
   FlameChart({
     @required this.data,
+    @required this.duration,
     @required this.flameChartWidth,
     @required this.flameChartHeight,
-  }) : timelineGrid = TimelineGrid(data.time.duration, flameChartWidth) {
-    _initRows();
+  }) : timelineGrid = TimelineGrid(duration, flameChartWidth) {
+    initRows();
   }
 
-  static const stackFramePadding = 1;
+  final T data;
 
-  final CpuProfileData data;
+  final Duration duration;
 
   // These values are not final because the flame chart viewport can change in
   // size.
   double flameChartWidth;
   double flameChartHeight;
 
   double get flameChartWidthWithInsets =>
```

Comment:
https://journal.stuffwithstuff.com/2016/06/16/long-names-are-long/

This repeats known context. How about `widthWithInsets`

Similar throughout with other names repeating `flameChart`